aten/src/ATen/CMakeLists.txt

```diff
@@ -580,7 +580,22 @@ install(FILES "${CMAKE_CURRENT_BINARY_DIR}/cmake-exports/ATenConfig.cmake"
 
 set(INSTALL_HEADERS ${base_h} ${ATen_CORE_HEADERS} ${native_nested_h} ${ATen_TRANSFORMER_HEADERS})
 if(NOT INTERN_BUILD_MOBILE)
-  list(APPEND INSTALL_HEADERS ${native_h} ${native_cpu_h} ${native_ao_sparse_h} ${native_quantized_h} ${cuda_h} ${native_cuda_h} ${native_hip_h} ${cudnn_h} ${hip_h} ${xpu_h} ${mps_h} ${native_mps_h} ${native_utils_h} ${miopen_h} ${mkldnn_xpu_h})
+  list(APPEND INSTALL_HEADERS ${native_h} ${native_cpu_h} ${native_ao_sparse_h} ${native_quantized_h} ${native_utils_h} ${miopen_h})
+  if(USE_CUDA)
+    list(APPEND INSTALL_HEADERS ${cuda_h} ${native_cuda_h})
+    if(CAFFE2_USE_CUDNN)
+      list(APPEND INSTALL_HEADERS ${cudnn_h})
+    endif()
+  endif()
+  if(USE_ROCM)
+    list(APPEND INSTALL_HEADERS ${hip_h} ${native_hip_h})
+  endif()
+  if(USE_MPS)
+    list(APPEND INSTALL_HEADERS ${mps_h} ${native_mps_h})
+  endif()
+  if(USE_XPU)
+    list(APPEND INSTALL_HEADERS ${xpu_h} ${mkldnn_xpu_h})
+  endif()
```

Comment:
Is it possible to refine the `IF-ELSE` block a little bit? It could be somthing like

```cmake
macro(APPEND_IF condition)
  list(GET ARGN 0 files)
  
  if(${condition})
    list(APPEND INSTALL_HEADERS ${files})
  endif()
endmacro()

APPEND_IF(USE_CUDA "${cuda_h} ${native_cuda_h}")
APPEND_IF(CAFFE2_USE_CUDNN "${cudnn_h}")
APPEND_IF(USE_ROCM "${hip_h} ${native_hip_h}")
APPEND_IF(USE_MPS "${mps_h} ${native_mps_h}")
APPEND_IF(USE_XPU "${xpu_h} ${mkldnn_xpu_h}")
```

Reply:
Good idea!